File: ehe/ehe.Web/Modules/DosyaUrunler/Urunler/UrunlerForm.cs
using Serenity.ComponentModel;

namespace ehe.DosyaUrunler.Forms;



[FormScript("DosyaUrunler.Urunler")] 
[BasedOnRow(typeof(UrunlerRow), CheckNames = true)]
[ColumnsScript]


public class UrunlerForm
{
    
    public string UrunAdi { get; set; }
    public string UrunAciklamasi { get; set; }
    public int UrunStokAdeti { get; set; }
    public int UrunFiyati { get; set; }
    public int UrunKodu { get; set; }
    public int UrunCesitleriid { get; set; }


    [IgnoreName, UrunYeniGrid]
    public string UrunYeniGrid { get; set; }







}

File: ehe/ehe.Web/Modules/ServerTypes/DosyaUrunler/UrunlerForm.ts
import { StringEditor, IntegerEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { UrunCesitleriDialog } from "../../DosyaUrunler/UrunCesitleri/UrunCesitleriDialog";
import { UrunYeniGrid } from "../../DosyaUrunler/Urunler/UrunYeniGrid";

export interface UrunlerForm {
    UrunAdi: StringEditor;
    UrunAciklamasi: StringEditor;
    UrunStokAdeti: StringEditor;
    UrunFiyati: IntegerEditor;
    UrunKodu: StringEditor;
    UrunCesitleriid: LookupEditor;
    UrunYeniGrid: UrunYeniGrid;
}

export class UrunlerForm extends PrefixedContext {
    static readonly formKey = 'DosyaUrunler.Urunler';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UrunlerForm.init)  {
            UrunlerForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = LookupEditor;
            var w3 = UrunYeniGrid;

            initFormType(UrunlerForm, [
                'UrunAdi', w0,
                'UrunAciklamasi', w0,
                'UrunStokAdeti', w0,
                'UrunFiyati', w1,
                'UrunKodu', w0,
                'UrunCesitleriid', w2,
                'UrunYeniGrid', w3
            ]);
        }
    }
}

queueMicrotask(() => [UrunCesitleriDialog]); // referenced dialogs